### ISSUE TITLE
No internet Access on UE

### ISSUE THREAD

**Issue by amintgan** [magma/magma#8147]
Hello @pshelar,

I have an on-prem-orc8r  and on-prem-agw, these two are connected as I can access the NMS dashboard. Also added the gateway with the subscriber details.
 The UE successful attach process was successful but there is no internet access. kindly find attached mme.logs
[magma logs.zip](https://github.com/magma/magma/files/6839452/magma.logs.zip)

**Comment by pshelar** [comment]
@amintgan as mentioned in the slack msg, I need DEBUG level logs to get more info from MME side. But posted logs are at info level.
```
000266 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1920        UDP log level........: INFO
000267 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1923        GTPV1-U log level....: INFO
000268 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1926        GTPV2-C log level....: INVALID_LOG_LEVEL
000269 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1929        SCTP log level.......: INFO
000270 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1932        S1AP log level.......: INFO
000271 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1935        ASN1 Verbosity level : 0
000272 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1938        NAS log level........: INFO
000273 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1941        MME_APP log level....: INFO
000274 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1944        SPGW_APP log level....: INFO
000275 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1947        S11 log level........: INVALID_LOG_LEVEL
000276 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1950        S6a log level........: INFO
000277 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1953        UTIL log level.......: INFO
000278 Sun Jul 11 10:09:13 2021 7F5DDE8B9BC0 INFO  CONFIG tasks/mme_app/mme_config.c      :1956        ITTI log level.......: ERROR (InTer-Task Interface)
```
please change log level in mme.yml. restart the MME. attache UE and post the losg.

**Comment by robinge330** [comment]
The change was done in /etc/magma/mme.yml to add log_level: DEBUG and process re-started. PLease could you check the mme3.log in the attached zip file which has DEBUG messages. Kindly let us know if we are missing any other DEBUG parameter

**Comment by pshelar** [comment]
can you post the mme.yml file.

**Comment by amintgan** [comment]
[magma Github troubleshooting.zip](https://github.com/magma/magma/files/6853807/magma.Github.troubleshooting.zip)
Please find attached mme1.yml file from directory (/etc/magma) and also mme1.conf from directory( /var/opt/magma/tmp).


Regards.

**Comment by pshelar** [comment]
@amintgan looks like OVS kernel module is not right.
can you post output of `modinfo openvswitch`

**Comment by amintgan** [comment]
Please in what directory do I find it?

**Comment by amintgan** [comment]
@pshelar please see output below:

magma@agw:~$ modinfo openvswitch
filename:       /lib/modules/5.8.0-59-generic/kernel/net/openvswitch/openvswitch.ko
alias:          net-pf-16-proto-16-family-ovs_ct_limit
alias:          net-pf-16-proto-16-family-ovs_meter
alias:          net-pf-16-proto-16-family-ovs_packet
alias:          net-pf-16-proto-16-family-ovs_flow
alias:          net-pf-16-proto-16-family-ovs_vport
alias:          net-pf-16-proto-16-family-ovs_datapath
license:        GPL
description:    Open vSwitch switching datapath
srcversion:     9B1A57AE22F7C4C5BB344DD
depends:        nf_conntrack,nf_nat,nf_conncount,libcrc32c,nf_defrag_ipv6,nsh
retpoline:      Y
intree:         Y
name:           openvswitch
vermagic:       5.8.0-59-generic SMP mod_unload modversions 
sig_id:         PKCS#7
signer:         Build time autogenerated kernel key
sig_key:        60:D1:39:00:A6:DA:91:43:D1:A1:22:AF:15:83:A2:98:A5:D3:5F:B4
sig_hashalgo:   sha512
signature:      9C:31:37:ED:E3:04:68:E4:65:3B:3D:4D:20:90:09:CD:C2:93:F2:37:
		09:04:0A:AE:11:CD:9F:CE:8C:3F:A8:F1:F4:0A:B0:6C:F0:4F:1B:FF:
		33:DE:AF:DC:44:AB:95:CC:73:04:59:4A:84:20:4F:3B:23:E3:0E:63:
		50:AA:60:43:2D:AB:F4:92:0B:AC:EE:22:EB:19:43:F0:C0:1C:5B:6E:
		F9:58:BE:00:4C:77:74:44:AD:25:6E:A6:48:A3:37:8F:E5:87:4F:2B:
		84:C3:1F:8D:A3:0C:B7:7B:50:16:D3:B6:31:B8:AE:80:F4:70:31:72:
		BF:B4:5F:C3:2B:6E:2A:E1:5D:C2:42:EC:59:EE:90:01:A0:0E:F2:E9:
		4F:61:B4:07:0D:6C:D8:6E:7E:F7:1F:08:C7:A8:C2:97:3D:30:84:D8:
		7C:C5:D5:14:75:A4:35:02:F7:54:DF:22:6F:C5:3C:C7:2B:E3:16:80:
		2E:A3:AF:40:FD:0F:FA:B9:94:4F:AD:A1:CE:A2:C5:C1:E3:40:84:08:
		36:41:E4:76:D8:5A:BB:FB:4A:67:78:78:91:A3:15:E5:E0:39:68:5A:
		47:1A:B0:09:0A:EC:29:86:96:C7:E9:6C:70:5B:04:9E:27:60:2B:E6:
		E7:1A:B9:9A:02:A3:D0:D0:00:6D:C9:5A:E1:3E:C4:77:8F:53:D0:B4:
		FE:20:09:D0:F4:08:AD:7E:99:55:A9:8A:D4:56:A4:9F:52:50:F9:35:
		36:25:5A:44:C6:B5:14:A3:1D:AE:EA:11:D1:90:4E:91:82:EA:26:5D:
		83:8F:F5:00:60:55:2A:EC:A5:16:D1:0C:43:B9:74:D9:19:8C:10:2F:
		9D:99:F6:10:AE:94:B0:5B:9E:D8:7E:2B:6D:28:6B:B2:67:D7:B4:F6:
		29:8F:C1:28:2A:5D:A2:B6:76:60:65:74:BD:6F:A9:1B:CC:6A:59:5B:
		E8:AD:53:65:E7:88:D6:32:E3:3E:7B:4E:2F:02:C3:F1:D3:FB:37:7B:
		79:CE:4C:E0:0B:D5:B7:C6:36:D3:A0:41:BC:63:86:85:7C:4D:35:9C:
		00:85:7C:D2:63:AA:28:19:5C:8B:19:1E:1C:DC:3B:EF:51:07:2A:4A:
		E7:DA:DA:C1:54:80:DC:21:F3:2B:ED:C3:BF:62:83:21:05:AF:5C:6E:
		CC:27:E4:52:02:0E:B7:28:67:B9:D0:01:08:0C:97:7A:AE:7B:8D:AF:
		D3:5D:E5:D2:9C:CB:A7:54:F3:96:19:91:12:5A:2D:AD:7B:52:66:10:
		A7:29:05:BE:17:EA:8B:4C:DB:88:90:72:D0:27:7E:4C:D0:ED:9E:AA:
		4F:B2:90:2E:BD:AD:CF:98:52:28:48:BB

**Comment by pshelar** [comment]
looks like the AGW does not have right kernel module, `ovs-kmod-upgrade.sh` on the AGW to get right OVS module installed.

**Comment by amintgan** [comment]
hello @pshelar the command returned an error as seen below: sudo ovs-kmod-upgrade.sh

**Building module:
cleaning build area...(bad exit status: 2)
./configure --with-linux='/lib/modules/5.8.0-59-generic/build' && make -C datapath/linux......(bad exit status: 1)
ERROR: Cannot create report: [Errno 17] File exists: '/var/crash/openvswitch-datapath-dkms.0.crash'
Error! Bad return status for module build on kernel: 5.8.0-59-generic (x86_64)
Consult /var/lib/dkms/openvswitch/2.14.3/build/make.log for more information.
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
run-parts: failed to stat component /etc/network/if-post-down.d/avahi-daemon: No such file or directory
 * Detected internal interfaces: 
 * Exiting ovsdb-server (1225498)
 * Starting ovsdb-server
 * Configuring Open vSwitch system IDs
 * Exiting ovs-vswitchd (1225518)
 * Saving interface configuration
 * Removing openvswitch module
 * Inserting openvswitch module
 * Starting ovs-vswitchd
 * Enabling remote OVSDB managers
 * Restoring interface configuration
 * Finding processes on dead interfaces
RTNETLINK answers: File exists
ovs-vsctl: Error detected while setting up 'gtp0': could not add network device gtp0 to ofproto (Address family not supported by protocol).  See ovs-vswitchd log for details.
ovs-vsctl: The default log directory is "/var/log/openvswitch".
net.ipv4.ip_forward = 1
modinfo: ERROR: Module /lib/modules/5.8.0-59-generic/updates/dkms/openvswitch.ko not found.
update failed
To cleanup control plane state run: config_stateless_agw.py flushall_redis**

**Comment by pshelar** [comment]
Release 1.6 does not support kernel 5.8, please downgrade it to 5.4.

**Comment by amintgan** [comment]
How do I downgrade to 1.5.4 ?

**Comment by sudhikan** [comment]
@amintgan - the kernel version required is 5.4.0-77. From the log messages above it seems you are running 5.8.0-59.

note that this is **NOT** a magma downgrade, instead a kernel version downgrade on the linux host. 

Found this [resource](http://www.humans-enabled.com/2014/08/how-to-set-default-grub-kernel-boot.html) that could help.

Additionally, it might be installing the older kernel version and then removing the existing followed by a reboot.

```
apt update
apt install -y python-minimal aptitude linux-image-5.4.0-77-generic linux-headers-5.4.0-77-generic
apt remove -y linux-image-5.8.0-59-generic
sudo reboot
```

cc: @tmdzk

**Comment by amintgan** [comment]
@sudhikan Thanks.

The kernel has been downgraded to 5.4.0-80-generic successfully, as 5.4.0-77-generic did not boot into os as it kept prompting "Busybox Initramfs Error".

Kindly advice on the next action

**Comment by sudhikan** [comment]
https://ostechnix.com/how-to-fix-busybox-initramfs-error-on-ubuntu/

**Comment by amintgan** [comment]
@sudhikan thanks.

**Comment by amintgan** [comment]
@sudhikan Hi, we have been able to complete the end-to-end test after downgrading the kernel version to 5.4.0-88 as 5.4.0-77 keeps prompting the busybox intitramsfs error even after following the troubleshooting link.

However, we have observed a new pattern as the S1AP port gets disconnected on the eNB after 3days of mme been connected. At every occasion this happens, a simple ifup eth0 resolves it.
